tutorial/write-custom-widget.md

```diff
@@ -107,3 +107,184 @@ Take care not to keep any important state in your renderer - animation tickers
 are well suited to that location but user state would not be. A widget that is
 hidden may have it's renderer destroyed and if it is shown again the new renderer
 must be able to reflect the same widget state.
+
+### Example Widget
+
+The following covers in detail the creation of a simple Widget and its Renderer.
+
+#### Create a Widget
+
+As described above, we need to inherit from the `widget.BaseWidget` type.
+
+Go provides an simple mechanism for inheriting behaviour from other types. Adding an Anonymous type as a field in the widget structure means that the widget inherits the behaviour of that Anonymous type.
+
+This also means that your widget now implements all of the public methods and properties define by the  `widget.BaseWidget` type.
+
+We add an Anonymous `widget.BaseWidget` as follows:
+
+```go
+type MyWidget struct {
+	widget.BaseWidget           // Inherit from BaseWidget
+	text              string	// The text to display in the widget
```

Comment:
The alignment here is untidy - I think it is because there is a tab character after `string` so it is not lining up - it should be just spaces after content (tabs indent at the beginning only).